### PR TITLE
Add mlpack

### PR DESCRIPTION
**Pull request by marcelotrevisani**
Related to https://github.com/conda-forge/staged-recipes/issues/4333

### CODE REVIEW

File: recipes/mlpack/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    - {{ compiler('cxx') }}
+    - cmake
+    - setuptools
+    - cython
+  host:
+    - boost
+    - python
+    - numpy
+    - pandas
+    - armadillo
```
Comment (mariusvniekerk) [inline recipes/mlpack/meta.yaml]: i'd move `cython` and `pip` to this section.  cython is tied to your python version

Comment (marcelotrevisani) [reply recipes/mlpack/meta.yaml]: Got it!

**Comment by marcelotrevisani** [comment]
I don't why ``conda build`` is trying to use ``gcc 4.8`` instead of the ``gcc 7.2``

@mariusvniekerk do you have any clue here?

**Comment by marcelotrevisani** [comment]
Please review, @conda-forge/staged-recipes

File: recipes/mlpack/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    - cmake
+  host:
+    - cython
+    - pip
+    - boost
+    - python
+    - numpy
+    - pandas  > 0.15.0
+    - armadillo
+    - pytest-runner
```
Comment (mariusvniekerk) [inline recipes/mlpack/meta.yaml]: Is this a build time dependency?

Comment (marcelotrevisani) [reply recipes/mlpack/meta.yaml]: Yes, I tried without it. However, their building process need the ``pytest-runner`` at build time

File: recipes/mlpack/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
+    - cmake
+  host:
+    - cython
+    - pip
+    - boost
+    - python
+    - numpy
+    - pandas  > 0.15.0
```
[the remark this answers is not in the document]
Comment (marcelotrevisani) [reply recipes/mlpack/meta.yaml]: Sure! Do I have to revert the merge commit? or can I do this change after the repository is created?

[the remark this answers is not in the document]
Comment (scopatz) [reply recipes/mlpack/meta.yaml]: You can change it after the feedstock repo is created!

[the remark this answers is not in the document]
Comment (isuruf) [reply recipes/mlpack/meta.yaml]: AFAIK, feedstock will not be created until this is fixed

[the remark this answers is not in the document]
Comment (scopatz) [reply recipes/mlpack/meta.yaml]: Please put in another PR @marcelotrevisani, thanks!

**Comment by marcelotrevisani** [comment]
The ``mlpack`` was not created yet, is something wrong?

**Pull request merged**